src/sentry/spans/buffer.py

```diff
@@ -200,36 +193,58 @@ def process_spans(self, spans: Sequence[Span], now: int):
 
                     queue_keys.append(queue_key)
 
-                results = iter(p.execute())
-                for hole_size, delete_item, item, has_root_span in results:
-                    # For each span, hole_size measures how long it took to
-                    # find the corresponding intermediate segment. Larger
-                    # numbers loosely correlate with fewer siblings per tree
-                    # level.
-                    min_hole_size = min(min_hole_size, hole_size)
-                    max_hole_size = max(max_hole_size, hole_size)
-                    queue_delete_items.append(delete_item)
-                    queue_items.append(item)
-                    queue_item_has_root_span.append(has_root_span)
+                results = p.execute()
 
         with metrics.timer("spans.buffer.process_spans.update_queue"):
-            with self.client.pipeline(transaction=False) as p:
-                for key, delete_item, item, has_root_span in zip(
-                    queue_keys, queue_delete_items, queue_items, queue_item_has_root_span
-                ):
-                    # if the currently processed span is a root span, OR the buffer
-                    # already had a root span inside, use a different timeout than
-                    # usual.
-                    if has_root_span:
-                        has_root_span_count += 1
-                        timestamp = now + self.span_buffer_root_timeout_secs
-                    else:
-                        timestamp = now + self.span_buffer_timeout_secs
+            queue_delete_items: dict[str, set[bytes]] = {}
+            queue_add_items: dict[str, MutableMapping[str | bytes, int]] = {}
 
-                    if delete_item != item:
-                        p.zrem(key, delete_item)
-                    p.zadd(key, {item: timestamp})
-                    p.expire(key, self.redis_ttl)
+            has_root_span_count = 0
+            min_hole_size = float("inf")
+            max_hole_size = float("-inf")
```

Comment:
Note I'm renaming these in https://github.com/getsentry/sentry/pull/88453, so we will need to resolve a conflict there.